samples/nrf9160/nrf_cloud_rest_device_message/src/main.c

```diff
@@ -0,0 +1,429 @@
+	case LTE_LC_EVT_NW_REG_STATUS:
+		LOG_DBG("LTE_LC_EVT_NW_REG_STATUS: %d", evt->nw_reg_status);
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			break;
+		}
+
+		LOG_DBG("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_DBG("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		break;
+	default:
+		break;
+	}
+}
+
+static int connect_to_lte(void)
+{
+	int err;
+
+	LOG_INF("Waiting for network...");
+
+	k_sem_reset(&lte_connected);
+
+	err = lte_lc_init_and_connect_async(lte_handler);
+	if (err) {
+		LOG_ERR("Failed to init modem, error: %d", err);
+		return err;
+	}
+
+	k_sem_take(&lte_connected, K_FOREVER);
+	(void)set_led(1);
+	LOG_INF("Connected to LTE");
+	return 0;
+}
+
+
+int setup_connection(void)
+{
+	/* Get the device ID */
+	int err = nrf_cloud_client_id_get(device_id, sizeof(device_id));
+
+	if (err) {
+		LOG_ERR("Failed to get device ID, error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Device ID: %s", log_strdup(device_id));
+
+	/* Connect to LTE */
+	err = connect_to_lte();
+
+	if (err) {
+		LOG_ERR("Failed to connect to cellular network: %d", err);
+		return err;
+	}
+
+	/* Wait until we know what time it is (necessary for JSON Web Token generation) */
+	err = modem_time_wait();
+
+	if (err) {
+		LOG_ERR("Modem failed to acquire real time: %d", err);
+		return err;
+	}
+
+	/* Perform JITP if enabled and requested */
+	if (IS_ENABLED(CONFIG_REST_DEVICE_MESSAGE_DO_JITP) && jitp_requested) {
+		err = do_jitp();
+
+		if (err) {
+			LOG_ERR("Failed to perform JITP, %d", err);
+			return err;
+		}
+	}
+
+	/* Update our shadow via REST */
+	err = update_shadow();
+	if (err) {
+		LOG_ERR("Failed to update device shadow via REST, %d", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static void offer_jitp(void)
+{
+	int ret;
+
+	jitp_requested = false;
+
+	k_sem_reset(&button_press_sem);
+
+	LOG_INF("---> Press button %d to request just-in-time provisioning", BTN_NUM);
+	LOG_INF("     Waiting %d seconds...", JITP_REQ_WAIT_SEC);
+
+	ret = k_sem_take(&button_press_sem, K_SECONDS(JITP_REQ_WAIT_SEC));
+	if (ret == 0) {
+		jitp_requested = true;
+		LOG_INF("JITP will be performed after network connection is obtained");
+	} else {
+		if (ret != -EAGAIN) {
+			LOG_ERR("k_sem_take error: %d", ret);
+		}
+
+		LOG_INF("JITP will be skipped");
+	}
+}
+
+int init(void)
+{
+	/* Init the LED */
+	int err = init_led();
+
+	if (err) {
+		LOG_ERR("LED initialization failed");
+		return err;
+	}
+
+	/* Init modem and print some status info */
+	int modem_lib_init_result = nrf_modem_lib_init(NORMAL_MODE);
+
+	if (modem_lib_init_result) {
+		LOG_ERR("Failed to initialize modem library: 0x%X", modem_lib_init_result);
+		return -EFAULT;
+	}
+
+	err = modem_info_init();
```

Comment:
for simplicity, you could omit printing the info/version/names...
this was done in the FOTA sample because the values would be different after a FOTA update.

Reply:
Good point, I'll do that.